src/components/ha-selector/ha-selector-number.ts

```diff
@@ -64,7 +64,7 @@ export class HaNumberSelector extends LitElement {
           class=${classMap({ single: this.selector.number?.mode === "box" })}
           .min=${this.selector.number?.min}
           .max=${this.selector.number?.max}
-          .value=${this.value ?? ""}
+          .value=${Object.is(this.value, -0) ? "-0" : this.value ?? ""}
```

Comment:
Too much of a hack IMO.  We should fix this more generically in `ha-text-field`, unless it only occurs when using `ha-selector-number`?

Reply:
I am not sure if it is possible to fix in ha-textfield. 

ha-textfield extends mwc-textfield, where `value` is a `string` property. 

The issue is that when user types `"-0"`, we get a event with value `"-0"` from mwc-textfield and convert it to Number, which is `-0`.

Then when we next render the ha-textfield, we set value to `-0`. But since value is a string, the number is implicitly converted to string. Unfortunately `Number(-0).toString()` is `"0"`, not `"-0"`. 

Without changing ha-textfield to accept number types, it never sees the `-0` input. 

I'm not sure if there is such a thing as a "more generic" fix, since I believe 0/-0 are the only numbers that have this issue.

Reply:
I guess if there was an alternative fix, it would be ha-selector-number storing the value from the textfield as a string instead of a number. Haven't thought through the implications of that though.

Reply:
I guess I was assuming that the issue was caused by autocorrection in `ha-text-field`.  If that's not the case and it's instead caused by string coercion, then we should better control that and get rid of the type incompatibility.

Reply:
Ok I've uploaded an alternative fix that I think I like better, it fixes more cases of text field changing value unexpectedly. 

selector now tracks the string in the textbox in addition to the number provided by the property, and only overwrites the string when converting that string to a number gives a different value than the number provided by the property. 

Let me know if that looks better.